controllers/user.js

```diff
@@ -594,6 +685,1369 @@ function logout(req, res) {
+    resolutionTimeData: null,
+  };
+  let kidArr;
+  if (req.query && req.query.from_file) {
+    //read from static file
+    const kidFile = await readFile("kids.txt", "binary");
+    kidArr = kidFile.toString().split("\n");
+  } else {
+    //read from DB
+    kidArr = await DB.getRemoveParticipants();
+  }
+
+  for await (const kid of kidArr) {
+    const userStats = await getRemoveRateByKid(kid, true); //true = aggregate performance metrics, skipping individual resolution time calculations
+    //console.log("userStats", userStats, kid);
+    if (userStats) {
+      if (userStats.totalResults) {
+        allStats.totalResults += userStats.totalResults;
+      }
+      if (userStats.removedResults) {
+        allStats.removedResults += userStats.removedResults;
+      }
+      if (
+        userStats.resolutionTimeArray &&
+        userStats.resolutionTimeArray.length
+      ) {
+        allStats.resolutionTimeArray = [
+          ...allStats.resolutionTimeArray,
+          ...userStats.resolutionTimeArray,
+        ];
+      }
+    }
+  }
+
+  allStats.resolutionPct = await FormUtils.calculatePercentage(
+    allStats.removedResults,
+    allStats.totalResults
+  );
+  allStats.resolutionTimeData = await calculateAverageResolutionTime(
+    allStats.resolutionTimeArray
+  );
+
+  res.render("dashboards", {
+    title: req.fluentFormat("Firefox Monitor"),
+    csrfToken: req.csrfToken(),
+    stats: allStats,
+    whichPartial: "dashboards/remove-all-stats",
+  });
+}
+
+async function getRemoveStatsUser(req, res) {
+  //MH TODO: validate form data server side
+  if (!req.user) {
+    console.error("no user");
+    const localeError = LocaleUtils.fluentFormat(
+      req.supportedLocales,
+      "remove-error-no-user"
+    );
+    return res.status(404).json({
+      error: localeError,
+    });
+  }
+
+  const user = req.user;
+
+  if (!user.kid) {
+    console.error("no kid user", user);
+    const localeError = LocaleUtils.fluentFormat(
+      req.supportedLocales,
+      "remove-error-no-kid"
+    );
+    return res.status(404).json({
+      error: localeError,
+    });
+  }
+
+  const userStats = await getRemoveRateByKid(user.kid, false);
+
+  res.render("dashboards", {
+    title: req.fluentFormat("Firefox Monitor"),
+    csrfToken: req.csrfToken(),
+    stats: userStats,
+    styleNonce: res.locals.styleNonce,
+    whichPartial: "dashboards/remove-all-stats",
+  });
+}
+
+async function createRemoveHashWaitlist(req, res) {
+  if (!req.user) {
+    console.error("no user");
+    const localeError = LocaleUtils.fluentFormat(
+      req.supportedLocales,
+      "remove-error-no-user"
+    );
+    return res.status(404).json({
+      error: localeError,
+    });
+  }
+
+  const user = req.user;
+
+  if (!user.primary_email.includes("@mozilla.com")) {
+    console.error("non mozilla email");
+    return res.status(404).json({
+      error:
+        "You must be signed in with a mozilla.com email address to access this page",
+    });
+  }
```

Comment:
should fox-food-specific code be removed?

Reply:
This will actually be used throughout the pilot - but I'm going to try and arrange for Kanary to give us this data via an SQL query from their own DB rather than making an API request for all of this. I'll add a ticket to have this removed.